lib/krane/deprecated_deploy_task.rb

```diff
@@ -447,6 +438,31 @@ def deploy_resources(resources, prune: false, verify:, record_summary: true)
       end
     end
 
+    def replace_or_create_resource!(resource, force: false)
+      args = if force
+        ["replace", "--force", "--cascade", "-f", resource.file_path]
+      else
+        ["replace", "-f", resource.file_path]
+      end
+
+      _, _, status = kubectl.run(args, log_failure: false)
+      return if status.success?
+      # it doesn't exist so we can't replace it, we try to create it
+      create_resource!(resource)
+    end
+
+    def create_resource!(resource)
+      out, err, status = kubectl.run("create", "-f", resource.file_path, log_failure: false, output: 'json')
+
+      resource.use_generate_name(out) if resource.deploy_method == :create
+      return if status.success?
+
+      raise FatalDeploymentError, <<~MSG
+        Failed to replace or create resource: #{resource.id}
+        #{err}
+      MSG
```

Comment:
I feel like the caller should be responsible for catching this error (e.g. in`deploy_resources`)